### PR TITLE
Fix scaler crash

### LINKED ISSUE #397: Scaler component restart while multiple HttpScaledObjects
### Report

When multiple services (usually >5) are deployed with HttpScaledObjects, scaling stops working properly. This looks to be caused by scaler component restarts.

### Expected Behavior

Scaling should be working with any number of depoyments/HttpScaledObjects.

### Actual Behavior

Scaling takes long time to happen or doesn't happen at all(issue is worse with higher number of deployments)

### Steps to Reproduce the Problem

1. Prepare deployment of service which is using scaling provided by http-add-on
2. Deploy same service multiple times to k8s cluster
3. Try to connect to one of services

### Logs from KEDA HTTP operator

Log from scaler component

[keda-add-ons-http-external-scaler.log](https://github.com/kedacore/http-add-on/files/8057911/keda-add-ons-http-external-scaler.log)

### What version of the KEDA HTTP Addon are you running?

0.3.0

### Kubernetes Version

1.21

### Platform

Amazon Web Services

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by leska-j**
Fixes #397
fixes issue by replacing merge of counts and routing table entries with lookup,
there is second alternative to this [PR](https://github.com/kedacore/http-add-on/pull/398), please chose one which is better

Closes #398

### CODE REVIEW

File: scaler/main.go [hunk 1]
```diff
@@ -184,7 +186,7 @@ func startAdminServer(
 	})
 	mux.HandleFunc("/queue", func(w http.ResponseWriter, r *http.Request) {
 		lggr = lggr.WithName("route.counts")
-		cts := pinger.counts()
+		cts := mergeCountsWithRoutingTable(pinger.counts(), routingTable)
```
Comment (arschles) [inline scaler/main.go:189]: this endpoint only needs to return the current state of the queue, not the state of the queue merged with the routing table. I think we can keep it as-is.

**Review by arschles — CHANGES_REQUESTED** [review]
@leska-j thanks for this one too. I think it looks like a better option than #398. Do you have a preference? Also, I left one request for a change here.

File: scaler/main.go (hunk 1 above)
Comment (leska-j) [reply scaler/main.go:189]: does this apply to the /queue_ping as well?

**Comment by leska-j** [comment]
> @leska-j thanks for this one too. I think it looks like a better option than #398. Do you have a preference? Also, I left one request for a change here.

Hi, i prefere this one as well, I provided second one just in case that merged count are required by some planned functionality

File: scaler/main.go (hunk 1 above)
Comment (arschles) [reply scaler/main.go]: >does this apply to the "/queue_ping" as well?

yes

>for now I removed it from there as well

thank you :)

**Review by arschles — APPROVED** [review]
@leska-j this looks great now. There is [one test failing](https://github.com/kedacore/http-add-on/runs/5205538452?check_suite_focus=true#step:4:86) still, but once you fix that we can merge this.

**Comment by leska-j** [comment]
@arschles thanks for review & merge

**Comment by arschles** [comment]
you're welcome @leska-j. feel free to contribute further, if you would like.

**Pull request merged**